v1/src/lib.rs

```diff
@@ -381,17 +407,18 @@ impl InfluxDbStorage {
         }
     }
 
-    async fn schedule_measurement_drop(&self, measurement: &str) -> TimedHandle {
-        let event = TimedEvent::once(
-            Instant::now() + Duration::from_millis(DROP_MEASUREMENT_TIMEOUT_MS),
-            TimedMeasurementDrop {
-                client: self.admin_client.clone(),
-                measurement: measurement.to_string(),
-            },
-        );
-        let handle = event.get_handle();
-        self.timer.add_async(event).await;
-        handle
+    async fn schedule_measurement_drop(&self, measurement: &str) -> ZResult<()> {
+        let m_string = measurement.to_string();
+        let cloned_client = self.client.clone();
+        await_task! {
+            {
+                if let Err(_) = tokio::time::timeout(Duration::from_millis(DROP_MEASUREMENT_TIMEOUT_MS), std::future::pending::<u8>()).await {
```

Comment:
It seems like it will block here instead of trigger the task in the future without blocking?

Reply:
If yes, I would recommend that we don't need to use `await_task`. We can have something like this

```rust
#[macro_export]
macro_rules! spawn_task {
    ($e: expr) => {
        match tokio::runtime::Handle::try_current() {
            Ok(handle) => {
                handle.spawn($e)
            },
            Err(_) => {
                TOKIO_RUNTIME.spawn($e)
            },
        };
    };
}
...
spawn_task!(async {
    // Maybe using tick is better
    tokio::time::sleep(Duration::from_millis(DROP_MEASUREMENT_TIMEOUT_MS)).await;
    drop_measurement(m_string, cloned_client).await;
})
```

Might have some syntax error (I didn't check)

Reply:
you are right, better to spawn and forget about task, seeing as it does not produce a result